index.js

```diff
@@ -42,114 +29,40 @@ const {
-  let lastMonthSheet = _.find(sheets, sheet => sheet.properties.title === lastMonthSheetTitle);
+  switch (transactionProvider) {
+    case 'plaid':
+      [currentMonthTransactions, lastMonthTransactions] = await getTransactions(
+        transactionColumns,
+        categoryOverrides,
+        currentMonthSheetTitle,
+      );
+      break;
 
-  if (!lastMonthSheet) {
-    const publicTemplateSheets = await getSheets(publicTemplateSheetId);
-    lastMonthSheet = await duplicateSheet(publicTemplateSheetId, publicTemplateSheets[0].properties.sheetId);
-    await renameSheet(lastMonthSheet.properties.sheetId, lastMonthSheetTitle);
-    await clearSheet(`${lastMonthSheetTitle}!${firstTransactionColumn}:${lastReferenceColumn}`);
+    default:
+      break;
   }
-  
-  if (!currentMonthSheet) {
-    currentMonthSheet = await duplicateSheet(process.env.SHEETS_SHEET_ID, lastMonthSheet.properties.sheetId);
-    await renameSheet(currentMonthSheet.properties.sheetId, currentMonthSheetTitle);
-    await clearSheet(`${currentMonthSheetTitle}!${firstTransactionColumn}:${lastReferenceColumn}`);
-  }
-
-  await clearSheet(`${currentMonthSheetTitle}!${firstTransactionColumn}:${lastTransactionColumn}`);
-  await clearSheet(`${lastMonthSheetTitle}!${firstTransactionColumn}:${lastTransactionColumn}`);
-
-  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-  // FETCH TRANSACTIONS AND MAP TO SHEET UPDATES
-  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-
-  const sanitizeTransaction = transaction => {
-    /* 
-    * Explode out Plaid's object hierarchy.
-    * 
-    * For example, the Category hierarchy comes as a list,
-    * and the first two are usually the only interesting ones.
-    * 
-    * Using defaultTransactionColumns above and _.get():
-    * 
-    *    { "category": ["Food and Drink", "Restaurants"] }
-    *
-    * would get expanded to:
-    * 
-    *    { "category.0": "Food and Drink", "category.1": "Restaurants" }
-    */
-    _.forEach(transactionColumns, column => {
-      transaction[column] = _.get(transaction, column);
-    })
-
-    // Map TRUE to 'y' and FALSE to nothing (used for Pending column)
-    transaction = _.mapValues(transaction, property => {
-      property = property === true ? 'y' : property;
-      property = property === false ? '' : property;
-      return property;
-    });
 
-    // Handle category overrides defined in .env
-    _.forEach(categoryOverrides, override => {
-      if (new RegExp(override.pattern, _.get(override, 'flags', '')).test(transaction.name)) {
-        transaction['category.0'] = _.get(override, 'category.0', '');
-        transaction['category.1'] = _.get(override, 'category.1', '');
-      }
-    });
-
-    return _.at(transaction, transactionColumns)
+  switch (spreadsheetProvider) {
+    case 'sheets':
+      await updateSheets(
```

Comment:
I'm not sure if there's a better way to do this in Node – looks kind of ugly with all these constants passed through like this.

I think these would be things we'd want defined in `index.js` to pass to other providers, though.

Reply:
I think its fine - ideally when we add additional providers the interface should be the same - that way we don't have to pass everything in for every additional provider, we can just have one call where we can change the method.